File: 6-23-剑指05-替换空格/replaceSpace.py
#直接遍历，遇见空格就将其替换为%20
class Solution:
    def replaceSpace(self, s: str) -> str:
        out = ""
        for i in range(0, len(s)):
            if s[i] != ' ':
                out = out + s[i]
            else:
                out = out + '%20'
        return out

obj = Solution()
s = "We are happy."
print(obj.replaceSpace(s))

File: Java/src/剑指OFFER/5-替换空格/ReplaceSpace.java
public class ReplaceSpace {

    public static void main(String[] args) {
        System.out.println("solution");

    }

    public static String replaceSpace(String s) {
        StringBuilder str = new StringBuilder();
        for (int i=0; i<s.length(); i++){
            if(s.charAt(i) == ' ')
                str.append("%20");
            else
                str.append(s.charAt(i));
        }
        return str.toString();
    }
}
